File: go/apis/slo/v1/elasticity_strategy_kind.go
package v1

// +kubebuilder:validation:Required

// ElasticityStrategyKind references a particular kind of ElasticityStrategy that should be used by an SloMapping.
type ElasticityStrategyKind struct {

	// The API group and version of the ElasticityStrategy.
	APIVersion string `json:"apiVersion"`

	// The kind of ElasticityStrategy.
	Kind string `json:"kind"`
}

File: ts/apps/slo/cpu-usage-slo-controller/src/app/model/sloc-crds/elasticity-strategy-kind.ts

/**
 * References a particular kind of ElasticityStrategy that should be used by an SloMapping.
 */
export interface ElasticityStrategyKind {

    /** The API group and version of the ElasticityStrategy. */
    apiVersion: string;

    /** The kind of ElasticityStrategy. */
    kind: string;

}
